Framework/DataHandling/test/CompressEventsTest.h

```diff
@@ -128,4 +127,74 @@ class CompressEventsTest : public CxxTest::TestSuite {
+    EventWorkspace_sptr input, output;
+    EventType eventType = WEIGHTED_NOTIME;
+    if (wallClockTolerance > 0.)
+      eventType = WEIGHTED;
+
+    /** Create event workspace with:
+     * 1 pixels (or another number)
+     * 64 histogrammed bins from 0.0 in steps of 1.0
+     * 128 events; two in each bin, at time 1.0, 2.0, etc.
+     * PulseTime = 1 second, 2 seconds, etc.
+     */
+    input = WorkspaceCreationHelper::createEventWorkspace(1, 64, 64, 0, 1, 2);
+    AnalysisDataService::Instance().addOrReplace("CompressEvents_input", input);
+
+    TS_ASSERT_EQUALS(input->getNumberEvents(), 128);
+    const double inputIntegral = input->getSpectrum(0).integrate(0., 100., true);
+
+    CompressEvents alg;
+    alg.initialize();
+    alg.setPropertyValue("InputWorkspace", "CompressEvents_input");
+    alg.setPropertyValue("OutputWorkspace", "CompressEvents_output");
+    alg.setProperty("Tolerance", tolerance);
+    alg.setPropertyValue("BinningMode", binningMode);
+    if (wallClockTolerance > 0.) {
+      alg.setProperty("WallClockTolerance", wallClockTolerance);
+      alg.setProperty("StartTime",
+                      "2010-01-01T00:00:00"); // copied from createEventWorkspace
+    }
+
+    TS_ASSERT_THROWS_NOTHING(alg.execute());
+    TS_ASSERT(alg.isExecuted());
+
+    output = AnalysisDataService::Instance().retrieveWS<EventWorkspace>("CompressEvents_output");
+
+    TS_ASSERT_EQUALS(output->getNumberEvents(), 7);
+    TS_ASSERT_EQUALS(output->getEventType(), eventType);
+    TS_ASSERT_DELTA(output->getSpectrum(0).integrate(0., 100., true), inputIntegral, 1.e-6);
+
+    EventList el = output->getSpectrum(0);
+    TS_ASSERT_DELTA(el.getEvent(0).weight(), 2.0, 1e-6);
+    TS_ASSERT_DELTA(el.getEvent(0).errorSquared(), 2.0, 1e-6);
+    TS_ASSERT_DELTA(el.getEvent(0).tof(), 0.5, 1e-6);
+    for (int i = 1; i < 7; i++) {
+      TS_ASSERT_DELTA(el.getEvent(i).weight(), 1.0 * pow(2, i), 1e-6);
+      TS_ASSERT_DELTA(el.getEvent(i).errorSquared(), 1.0 * pow(2, i), 1e-6);
+      TS_ASSERT_DELTA(el.getEvent(i).tof(), 0.75 * pow(2, i), 1e-6);
+    }
+
+    if (wallClockTolerance > 0.) {
+      int64_t firstTime = 631152000000000000;
+      TS_ASSERT_EQUALS(el.getEvent(0).pulseTime().totalNanoseconds(), firstTime);
+      TS_ASSERT_EQUALS(el.getEvent(1).pulseTime().totalNanoseconds(), firstTime + 1000000000);
+      TS_ASSERT_EQUALS(el.getEvent(2).pulseTime().totalNanoseconds(), firstTime + 2500000000);
+      TS_ASSERT_EQUALS(el.getEvent(3).pulseTime().totalNanoseconds(), firstTime + 5500000000);
+      TS_ASSERT_EQUALS(el.getEvent(4).pulseTime().totalNanoseconds(), firstTime + 11500000000);
+      TS_ASSERT_EQUALS(el.getEvent(5).pulseTime().totalNanoseconds(), firstTime + 23500000000);
+      TS_ASSERT_EQUALS(el.getEvent(6).pulseTime().totalNanoseconds(), firstTime + 47500000000);
+    } else {
+      for (int i = 0; i < 7; i++) {
+        TS_ASSERT_EQUALS(el.getEvent(i).pulseTime().totalNanoseconds(), 0);
```

Comment:
They are not the same. `defaultTime` is returning the unix epoch but `DateAndTime` is relative to the GPS epoch. So the difference between `DateAndTime::defaultTime()` and `DateAndTime(0)` is 20 years. I will update this to compare DateAndTime instead of nanoseconds as it makes it easier to understand.